### PR TITLE
Adding delete button with functionality to each list item

### LINKED ISSUE #11: 11. As a user, I want to be able to delete items from my shopping list so that my list isn’t cluttered with items I don’t want to buy in the future.
## Summary
Users might make a mistake when entering an item, or may decide they won’t be buying a certain item again in the future. Allowing them to delete list items will help them keep their lists tidy.
 
## Acceptance criteria
- [x] The `ListItem` component renders a button that allows the user to delete an item from their list when clicked
- [x] Clicking the delete button prompts the user to confirm that they *really* want to delete the item
- [x]  The `deleteItem` function in `api/firebase.js` has been filled out, and deletes the item from the Firestore database

## Notes:
* As noted in [the wireframe](https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/list-populated.png), activating the “Delete” button should show a confirmation dialog. We want to give users a chance to confirm “destructive” actions.
* It might be tempting to create your own modal dialog component to handle confirmation, but modals are fraught with accessibility concerns, so consider using a simple [JavaScript confirm dialog](https://developer.mozilla.org/en-US/docs/Web/API/Window/confirm).

### PR DESCRIPTION
**Pull request by itsoliviasparks**
## Description

- Rendered a delete button next to each list item
- On click of the delete button a confirmation window pops up to confirm the user wants to delete the item
- On confirmation the `deleteItem` function is called, and the item is deleted from firebase

## Related Issue

closes #11 

## Acceptance Criteria
- [x] The `ListItem` component renders a button that allows the user to delete an item from their list when clicked
- [x] Clicking the delete button prompts the user to confirm that they *really* want to delete the item
- [x]  The `deleteItem` function in `api/firebase.js` has been filled out, and deletes the item from the Firestore database

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### Before

Screenshot 2023-05-10 at 2 54 58 PM

### After

ezgif com-video-to-gif

## Testing Steps / QA Criteria

- open firebase to see items being deleted
- in the `List` view, click the delete button & confirm in the window to see the item delete in firebase

### CODE REVIEW

File: src/components/ListItem.jsx [hunk 1]
```diff
@@ -44,6 +44,13 @@ export function ListItem({
 		);
 	};
 
+	const handleDelete = (e) => {
+		e.preventDefault();
+		if (window.confirm('Are you sure you want to delete this item?')) {
+			deleteItem(listToken, itemId);
+		}
+	};
+
```
Comment (hansontram) [inline src/components/ListItem.jsx:53]: This `window.confirm` method is so cool! 😁

May be a bit extra but how do you feel about displaying the item in the message to highlight exactly which item the user is intending to delete? 

ex:
```
		if (window.confirm(`Are you sure you want to delete "${name}"?`)) {
			deleteItem(listToken, itemId);
		}
```

**Review by hansontram — APPROVED** [review]
Great work on implementing the delete button feature! 👍🏼 
The code is clean and meets all acceptance criteria. I'm excited to see the app coming together 🚀

File: src/components/ListItem.jsx (hunk 1 above)
Comment (itsoliviasparks) [reply src/components/ListItem.jsx:53]: LOVE!

**Review by 01001101CK — APPROVED** [review]
Awesome job! The code is clean and easy to navigate. I am looking forward to our final product!

**Comment by vividavila98** [comment]
This new feature is going to make the app sooo much easier to test! The code looks great & Hanson's suggestion of adding the item's name was a nice touch. Also tested on my end and works as expected :)

**Review by vividavila98 — APPROVED** [review]
This new feature is going to make the app sooo much easier to test! The code looks great & Hanson's suggestion of adding the item's name was a nice touch. Also tested on my end and works as expected :)

**Pull request merged**